generate/generate.go

```diff
@@ -129,47 +132,198 @@ func CheckStack(root string, stack stack.S) ([]string, error) {
 		return nil, fmt.Errorf("checking for outdated code: %v", err)
 	}
 
-	logger.Trace().Msg("Generating backend cfg code for stack.")
-
 	stackpath := stack.AbsPath()
 	stackMeta := stack.Meta()
+
+	outdatedBackendFiles, err := backendConfigOutdatedFiles(root, stackpath, stackMeta, globals, cfg)
+	if err != nil {
+		return nil, fmt.Errorf("checking for outdated backend config: %v", err)
+	}
+	outdated = append(outdated, outdatedBackendFiles...)
+
+	outdatedLocalsFiles, err := exportedLocalsOutdatedFiles(root, stackpath, stackMeta, globals, cfg)
+	if err != nil {
+		return nil, fmt.Errorf("checking for outdated exported locals: %v", err)
+	}
+	outdated = append(outdated, outdatedLocalsFiles...)
+
+	outdatedTerraformFiles, err := exportedTerraformOutdatedFiles(root, stackpath, stackMeta, globals, cfg)
+	if err != nil {
+		return nil, fmt.Errorf("checking for outdated exported terraform: %v", err)
+	}
+	outdated = append(outdated, outdatedTerraformFiles...)
+
+	sort.Strings(outdated)
+
+	return outdated, nil
+}
+
+func backendConfigOutdatedFiles(
+	root, stackpath string,
+	stackMeta stack.Metadata,
+	globals *terramate.Globals,
+	cfg StackCfg,
+) ([]string, error) {
+	logger := log.With().
+		Str("action", "generate.backendConfigOutdatedFiles()").
+		Str("root", root).
+		Str("stackpath", stackpath).
+		Logger()
+
+	logger.Trace().Msg("Generating backend cfg code for stack.")
+
 	genbackend, err := generateBackendCfgCode(root, stackpath, stackMeta, globals, stackpath)
 	if err != nil {
-		return nil, fmt.Errorf("checking for outdated code: %v", err)
+		return nil, err
 	}
 
 	stackBackendCfgFile := filepath.Join(stackpath, cfg.BackendCfgFilename)
 	currentbackend, err := loadGeneratedCode(stackBackendCfgFile)
 	if err != nil {
-		return nil, fmt.Errorf("checking for outdated code: %v", err)
+		return nil, err
 	}
 
 	logger.Trace().Msg("Checking for outdated backend cfg code on stack.")
 
 	if string(genbackend) != string(currentbackend) {
-		logger.Trace().Msg("Detected outdated backend config.")
-		outdated = append(outdated, cfg.BackendCfgFilename)
+		logger.Trace().Msg("Detected outdated backend cfg.")
+		return []string{cfg.BackendCfgFilename}, nil
+	}
+
+	logger.Trace().Msg("backend cfg is updated.")
+	return nil, nil
+}
+
+func exportedTerraformOutdatedFiles(
+	root, stackpath string,
+	stackMeta stack.Metadata,
+	globals *terramate.Globals,
+	cfg StackCfg,
+) ([]string, error) {
+	logger := log.With().
+		Str("action", "generate.exportedTerraformOutdatedFiles()").
+		Str("root", root).
+		Str("stackpath", stackpath).
+		Logger()
+
+	logger.Trace().Msg("Checking for outdated exported terraform code on stack.")
+
+	loadedStackTf, err := genhcl.Load(root, stackMeta, globals)
+	if err != nil {
+		return nil, err
+	}
+
+	logger.Trace().Msg("Loaded exported terraform code, checking")
+
+	outdated := []string{}
+
+	for name, tf := range loadedStackTf.ExportedCode() {
+		exportedTfFilename := name
```

Comment:
non-blocking: would it make sense to rename `name` to `exportedTfFilename` in the loop so we don't need to assign a new variable here?

Reply:
Indeed we can, this was a left over from a previous version of the code that used the name to pass to a function, I removed the function but I was refactoring quickly, so I just removed the function call..but now it doesn't make sense :laughing: . Nice catch :heart: